src/Domain/CoordinateMaps/Wedge3D.cpp

```diff
@@ -0,0 +1,353 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Domain/CoordinateMaps/Wedge3D.hpp"
+
+#include "DataStructures/MakeWithValue.hpp"
+#include "DataStructures/Tensor/EagerMath/DeterminantAndInverse.hpp"
+#include "ErrorHandling/Assert.hpp"
+#include "ErrorHandling/Error.hpp"
+#include "Utilities/ConstantExpressions.hpp"
+
+namespace CoordinateMaps {
+
+Wedge3D::Wedge3D(const double radius_of_other_surface,
+                 const double radius_of_spherical_surface,
+                 const Direction<3> direction_of_wedge,
+                 const double sphericity_of_other_surface) noexcept
+    : radius_of_other_surface_(radius_of_other_surface),
+      radius_of_spherical_surface_(radius_of_spherical_surface),
+      direction_of_wedge_(direction_of_wedge),
+      sphericity_of_other_surface_(sphericity_of_other_surface) {
+  ASSERT(radius_of_other_surface > 0,
+         "The radius of the other surface must be greater than zero.");
+  ASSERT(radius_of_spherical_surface > 0,
+         "The radius of the spherical surface must be greater than zero.");
+  ASSERT(sphericity_of_other_surface >= 0 and sphericity_of_other_surface <= 1,
+         "Inner sphericity must be between 0 and 1");
+  ASSERT(radius_of_other_surface < radius_of_spherical_surface or
+         radius_of_other_surface > sqrt(3.) * radius_of_spherical_surface *
+                                       (1 - sphericity_of_other_surface));
```

Comment:
I told you to solve the equation and not try to guess the answer.
`(1 + sphericity_of_other_surface * (sqrt(3.) - 1)) * radius_of_other_surface > sqrt(3.) * radius_of_spherical_surface`